### PR TITLE
Add io.github.gen2brain.cbconvert

### PR DESCRIPTION
**Pull request by gen2brain**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

The patch is to build a MuPDF shared library. Every distribution that ships it as a shared library has something similar.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: io.github.gen2brain.cbconvert.yml [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+      - --disable-x265
+      - --disable-tests
+      - --disable-examples
+    sources:
+      - type: archive
+        url: https://github.com/strukturag/libheif/releases/download/v1.15.2/libheif-1.15.2.tar.gz
+        sha256: 7a4c6077f45180926583e2087571371bdd9cb21b6e6fada85a6fbd544f26a0e2
+
+  - name: highway
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.github.gen2brain.cbconvert.yml:50]: cmake-ninja?

File: io.github.gen2brain.cbconvert.yml [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+      - -DHWY_ENABLE_TESTS=OFF
+      - -DHWY_ENABLE_EXAMPLES=OFF
+      - -DHWY_WARNINGS_ARE_ERRORS=OFF
+    sources:
+      - type: archive
+        url: https://github.com/google/highway/archive/refs/tags/1.0.7.tar.gz
+        sha256: 5434488108186c170a5e2fca5e3c9b6ef59a1caa4d520b008a9b8be6b8abe6c5
+
+  - name: libjxl
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.github.gen2brain.cbconvert.yml:62]: cmake-ninja?

File: io.github.gen2brain.cbconvert.yml (hunk 1 above)
Comment (gen2brain) [reply io.github.gen2brain.cbconvert.yml]: Changed to cmake-ninja.

File: io.github.gen2brain.cbconvert.yml (hunk 2 above)
Comment (gen2brain) [reply io.github.gen2brain.cbconvert.yml]: Changed to cmake-ninja.

**Comment by hfiguiere** [comment]
bot, build io.github.gen2brain.cbconvert

File: io.github.gen2brain.cbconvert.yml [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+cleanup:
+  - '*.a'
+  - '*.la'
+  - '/bin/Magick*'
+  - '/etc'
+  - '/include'
+  - '/lib/cmake'
+  - '/lib/pkgconfig'
+  - '/lib/ImageMagick*'
+  - '/share'
```
Comment (hfiguiere) [inline io.github.gen2brain.cbconvert.yml:27]: this delete things it shouldn't. including the desktop file, etc.

Comment (gen2brain) [reply io.github.gen2brain.cbconvert.yml:27]: Thanks, I thought that cleanup was for the build directory only, if you mean `/share` is wrong. What about the id that ends with `.cbconvert` and the command is `cbconvert-gui`, I think I saw that is a problem in some PRs.

Comment (hfiguiere) [reply io.github.gen2brain.cbconvert.yml:27]: > thought that cleanup was for the build directory only,

it's never  about it.

it's all about installed files.

**Comment by hfiguiere** [comment]
bot, build io.github.gen2brain.cbconvert

File: io.github.gen2brain.cbconvert.yml [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+  - name: CBconvert
+    buildsystem: simple
+    build-commands:
+      - cp cmd/cbconvert-gui/dist/linux/flatpak/modules.txt cmd/cbconvert-gui/vendor/
+      - cd cmd/cbconvert-gui && $GOROOT/bin/go build -mod=vendor -trimpath -tags "extlib portal" -ldflags "-s -w -X main.appVersion=1.0.2"
+      - install -Dm00755 cmd/cbconvert-gui/cbconvert-gui $FLATPAK_DEST/bin/cbconvert
+      - install -Dm00644 cmd/cbconvert-gui/dist/linux/cbconvert.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png
+      - install -Dm00644 cmd/cbconvert-gui/dist/linux/flatpak/$FLATPAK_ID.desktop $FLATPAK_DEST/share/applications/$FLATPAK_ID.desktop
+      - install -Dm00644 cmd/cbconvert-gui/dist/linux/flatpak/$FLATPAK_ID.metainfo.xml $FLATPAK_DEST/share/metainfo/$FLATPAK_ID.metainfo.xml
+      - install -Dm00644 cmd/cbconvert-gui/dist/linux/flatpak/$FLATPAK_ID.metainfo.xml $FLATPAK_DEST/share/appdata/$FLATPAK_ID.appdata.xml
```
Comment (hfiguiere) [inline io.github.gen2brain.cbconvert.yml:165]:
```suggestion
```

this is useless.

**Comment by hfiguiere** [comment]
that's not the error. it's quite explict

**Comment by hfiguiere** [comment]
```
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
```

**Comment by gen2brain** [comment]
> ```
> • tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
> ```

I also added content_rating, somehow I missed it. I would appreciate it if you could queue another build. Thanks.

**Comment by hfiguiere** [comment]
you can do that too

**Comment by gen2brain** [comment]
bot, build io.github.gen2brain.cbconvert

**Pull request closed without merging**